### PR TITLE
agent/template: add static_secret_render_interval configurable

### PR DESCRIPTION
**Pull request by jasonodonnell**
[Consul Template recently added a feature](https://github.com/hashicorp/consul-template/commit/dc2df7fda88f8bc6b54e73de038d4cb4a8e07a98) to allow operators to override the default lease duration when a secret has no lease/isn't renewable. This default (5m) can be now altered using the following agent config:

```hcl
template_config {
    static_secret_render_interval = 60
}
```

We'll need to add this configurable to both Vault K8s and Vault Helm.

### CODE REVIEW

**Comment by jasonodonnell** [comment]
In case it confuses anyone, `consul-template` is already updated in main so I did not need to update the dependency for this feature.

**Review by tvoran — COMMENTED** [review]
Can you update the docs here too?

File: command/agent/config/config.go [hunk 1]
```diff
@@ -119,7 +119,9 @@ type Sink struct {
 
 // TemplateConfig defines global behaviors around template
 type TemplateConfig struct {
-	ExitOnRetryFailure bool `hcl:"exit_on_retry_failure"`
+	ExitOnRetryFailure      bool          `hcl:"exit_on_retry_failure"`
+	DefaultLeaseDurationRaw interface{}   `hcl:"default_lease_duration"`
```
[the remark this answers is not in the document]
Comment (jasonodonnell) [reply command/agent/config/config.go:123]: Talked internally about this and decided to keep as is for parity with CT.

File: website/content/docs/agent/template-config.mdx [hunk 2]
```diff
@@ -52,3 +52,8 @@ The top level `template_config` block has the following configuration entries:
 - `exit_on_retry_failure` `(bool: false)` - This option configures Vault Agent
 to exit after it has exhausted its number of template retry attempts due to
 failures.
+
+`static_secret_render_interval` `(string or integer: 5m)` - If specified, configures
```
Comment (tvoran) [inline website/content/docs/agent/template-config.mdx:56]:
```suggestion
- `static_secret_render_interval` `(string or integer: 5m)` - If specified, configures
```

**Review by tvoran — APPROVED** [review]
Also a reminder to update the PR description 😃

File: command/agent/config/config.go [hunk 3]
```diff
@@ -119,7 +119,9 @@ type Sink struct {
 
 // TemplateConfig defines global behaviors around template
 type TemplateConfig struct {
-	ExitOnRetryFailure bool `hcl:"exit_on_retry_failure"`
+	ExitOnRetryFailure       bool          `hcl:"exit_on_retry_failure"`
+	StaticSecretRenderIntRaw interface{}   `hcl:"static_secret_render_interval"`
```
Comment (calvn) [inline command/agent/config/config.go:123]: Did we land/agree on settling with `nonleased_secret_render_interval` as the param name?

**Pull request merged**